packages/react-menu/src/components/MenuGroup/renderMenuGroup.tsx

```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+import { getSlots } from '@fluentui/react-utils';
+import { MenuGroupState } from './MenuGroup.types';
+import { MenuGroupContextProvider } from '../../menuGroupContext';
+
+/**
+ * Redefine the render function to add slots. Reuse the menugroup structure but add
+ * slots to children.
+ */
+export const renderMenuGroup = (state: MenuGroupState) => {
+  const { slots, slotProps } = getSlots(state);
+
+  return (
+    <MenuGroupContextProvider value={{ headerId: state.headerId }}>
```

Comment:
I have the same comment as before, let's leave a TODO to stabilize a Context or use Context Selector

Reply:
```suggestion
    // TODO introduce context selector pattern to avoid unnecessary rerenders
    <MenuGroupContextProvider value={{ headerId: state.headerId }}>
```